### PR TITLE
Add table aws_kinesis_ consumer. Closes #181

### LINKED ISSUE #181: Add table aws_kinesis_ consumer.
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.37.24/service/kinesis#ConsumerDescription.GoString

### PR DESCRIPTION
**Pull request by visiit**
# Integration test logs
<details>
  <summary>
 vishalkumar@Vishals-MacBook-Air aws-test % TURBOT_PROFILE=shaktiman TF_VAR_aws_profile=shaktiman ./tint.js tests/aws_kinesis_consumer
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_kinesis_consumer []

PRETEST: tests/aws_kinesis_consumer

TEST: tests/aws_kinesis_consumer
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_kinesis_stream.named_test_resource: Creating...
aws_kinesis_stream.named_test_resource: Still creating... [10s elapsed]
aws_kinesis_stream.named_test_resource: Still creating... [20s elapsed]
aws_kinesis_stream.named_test_resource: Still creating... [30s elapsed]
aws_kinesis_stream.named_test_resource: Still creating... [40s elapsed]
aws_kinesis_stream.named_test_resource: Creation complete after 43s [id=arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294]
null_resource.named_test_resource: Creating...
null_resource.named_test_resource: Provisioning with 'local-exec'...
null_resource.named_test_resource (local-exec): Executing: ["/bin/sh" "-c" "aws kinesis register-stream-consumer --stream-arn arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294 --consumer-name turbottest76294 --profile shaktiman --region  us-east-1"]
null_resource.named_test_resource (local-exec): {
null_resource.named_test_resource (local-exec):     "Consumer": {
null_resource.named_test_resource (local-exec):         "ConsumerName": "turbottest76294",
null_resource.named_test_resource (local-exec):         "ConsumerARN": "arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294/consumer/turbottest76294:1616067805",
null_resource.named_test_resource (local-exec):         "ConsumerStatus": "CREATING",
null_resource.named_test_resource (local-exec):         "ConsumerCreationTimestamp": "2021-03-18T17:13:25+05:30"
null_resource.named_test_resource (local-exec):     }
null_resource.named_test_resource (local-exec): }
null_resource.named_test_resource: Creation complete after 3s [id=8370802676283163321]
null_resource.output_test_resource: Creating...
null_resource.output_test_resource: Provisioning with 'local-exec'...
null_resource.output_test_resource (local-exec): Executing: ["/bin/sh" "-c" "aws kinesis list-stream-consumers --stream-arn arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294 --output json --profile shaktiman --region us-east-1 > /private/var/folders/xc/kxfp1nw938dgs1ypvslzdsyc0000gr/T/tests/aws_kinesis_consumer/terraform/test/kinesis_stream.json"]
null_resource.output_test_resource: Creation complete after 3s [id=4756568803058456602]
data.local_file.input: Refreshing state...

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_region = us-east-1
resource_aka = arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294/consumer/turbottest76294:1616067805
resource_name = turbottest76294
stream_aka = arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294

Running SQL query: test-get-query.sql
[
  {
    "consumer_arn": "arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294/consumer/turbottest76294:1616067805",
    "consumer_name": "turbottest76294",
    "consumer_status": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "stream_arn": "arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "consumer_arn": "arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294/consumer/turbottest76294:1616067805",
    "consumer_name": "turbottest76294"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:kinesis:us-east-1:986325076436:stream/turbottest76294/consumer/turbottest76294:1616067805"
    ],
    "region": "us-east-1",
    "title": "turbottest76294"
  }
]
✔ PASSED

POSTTEST: tests/aws_kinesis_consumer

TEARDOWN: tests/aws_kinesis_consumer

SUMMARY:

1/1 passed.
</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>
+----------------------+----------------------------------------------------------------------------------------------------+-----------------+
| consumer_name        | consumer_arn                                                                                       | consumer_status |
+----------------------+----------------------------------------------------------------------------------------------------+-----------------+
| turbot-test-consumer | arn:aws:kinesis:us-east-1:986325076436:stream/test-stream/consumer/turbot-test-consumer:1615896100 | ACTIVE          |
+----------------------+----------------------------------------------------------------------------------------------------+-----------------+
</summary>
  
```
select
  consumer_name,
  consumer_arn,
  consumer_status,
  stream_arn
from
  aws_kinesis_consumer;

select
  consumer_name,
  consumer_status,
  consumer_arn
from
  aws_kinesis_consumer
where consumer_status = 'CREATING' or consumer_status = 'DELETING';
```
</details>

### CODE REVIEW

File: aws-test/tests/aws_kinesis_consumer/variables.tf [hunk 1]
```diff
@@ -0,0 +1,99 @@
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20200125-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "aws_profile" {
+  type        = string
+  default     = "integration-tests"
```
Comment (khushboo9024) [inline aws-test/tests/aws_kinesis_consumer/variables.tf:9]:
```suggestion
  default     = "default"
```

File: aws/table_aws_kinesis_consumer.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listKinesisConsumer", "AWS_REGION", region)
+	streamData := *h.Item.(*kinesis.DescribeStreamOutput)
+
+	c, err := getCommonColumns(ctx, d, h)
+
+	commonColumnData := c.(*awsCommonColumnData)
+
+	arn := "arn:" + commonColumnData.Partition + ":kinesis:" + commonColumnData.Region + ":" + commonColumnData.AccountId + ":stream" + "/" + *streamData.StreamDescription.StreamName
```
Comment (khushboo9024) [inline aws/table_aws_kinesis_consumer.go:90]: can't we use `StreamARN` directly ?

File: aws/table_aws_kinesis_consumer.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	)
+
+	return nil, err
+}
+
+func getAwsKinesisConsumer(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsKinesisConsumer")
+
+
```
Comment (khushboo9024) [inline aws/table_aws_kinesis_consumer.go:117]:
```suggestion
```

File: aws-test/tests/aws_kinesis_consumer/variables.tf [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  provisioner "local-exec" {
+   command = "aws kinesis register-stream-consumer --stream-arn ${aws_kinesis_stream.named_test_resource.arn} --consumer-name ${var.resource_name} --profile ${var.aws_profile} --region  ${data.aws_region.primary.name}"
+  }
+}
+
+locals {
+  path = "${path.cwd}/kinesis_stream.json"
+}
+
+
```
Comment (khushboo9024) [inline aws-test/tests/aws_kinesis_consumer/variables.tf:67]:
```suggestion
```

**Review by khushboo9024 — COMMENTED** [review]
Please resolve the conflict.

**Comment by visiit** [comment]
Resolved in https://github.com/turbot/steampipe-plugin-aws/pull/222

**Pull request closed without merging**